### PR TITLE
moving to chrome 41

### PR DESCRIPTION
**Pull request by adambender**

### CODE REVIEW

File: lib/gen/push_messaging.dart [hunk 1]
```diff
@@ -1,8 +1,9 @@
 /* This file has been generated from push_messaging.idl - do not edit */
 
 /**
- * Use `chrome.pushMessaging` to enable apps and extensions to receive message
- * data sent through <a href="cloudMessaging.html">Google Cloud Messaging</a>.
+ * The `chrome.pushMessaging` API is deprecated since Chrome 38, and will no
+ * longer be supported in Chrome 41. Switch to `$(ref:gcm chrome.gcm)` to take
+ * advantage of <a href="cloudMessaging.html">Google Cloud Messaging</a>.
```
Comment (devoncarew) [inline lib/gen/push_messaging.dart]: Perhaps we should not longer generate this API?

**Comment by devoncarew** [comment]
lgtm!

One comment about a deprecated (and removed?) API. Let's file an issue to track that.

Also, we should update the changelog with info about the removed API (scriptBadge), and that the APIs have been upgraded to chrome 41.

It looks like there's a build failure (https://drone.io/github.com/dart-gde/chrome.dart/259). I think the `overrides.json` file might need to be tweaked to share some classes, or disambiguate two different classes with the same name.

Due to the size of this PR, it might be practical to land it, and fix the build issue and other items (changelog) in follow on PRS.

**Comment by adambender** [comment]
I have update the changelog and removed pushMessaging from generation since it is no longer available in M41. Did you want me to file an issue specifically for pushMessaging?

**Comment by adambender** [comment]
Also, it looks like there is a real problem in the context_menus.dart and tabs.dart resulting from the issue you saw in drone.io.

Would you mind adding me to the drone.io project so I can get a notification when it breaks?

**Comment by devoncarew** [comment]
> Did you want me to file an issue specifically for pushMessaging?

No need if you've already removed it.

> Would you mind adding me to the drone.io project so I can get a notification when it breaks?

Happy to - let me know your drone.io username. I think we'll want to move this over to building with travis though - it's more featureful at this point.

**Comment by adambender** [comment]
drone.io: adambender

File: lib/gen/context_menus.dart [hunk 2]
```diff
@@ -7,8 +7,8 @@
  */
 library chrome.contextMenus;
 
-//import 'context_menus_internal.dart';
```
Comment (adambender) [inline lib/gen/context_menus.dart]: I was searching through the diff and noticed this, were you previously hand-removing the internal reference here?

**Comment by adambender** [comment]
I was able to address the issues with extension_types by adding them to the apis.json, apparently they moved a couple thing from types.json to extension_types.json.

File: lib/gen/context_menus.dart (hunk 2 above)
Comment (devoncarew) [inline lib/gen/context_menus.dart]: Yes, sadly enough -

**Comment by devoncarew** [comment]
> I was able to address the issues with extension_types by adding them to the apis.json

Awesome!

re: drone.io, I think I'll just turn down that bot, as we have travis up and running now. Travis can do things like write the build status back into the PR as commit metadata - you know in the PR whether the build has passed, failed, or is still running.

**Comment by adambender** [comment]
Clearing up the extension types and context_menus_internal issue revealed a couple name collisions which I have now addressed. As far as I can tell, everything is building as intended. Any objections to my merging, cutting 0.7.0-dev1 and pushing to pub?

**Comment by devoncarew** [comment]
Sounds good! lgtm

**Comment by adambender** [comment]
I may need access granted for pub :(

**Pull request merged**